src/huggingface_hub/_inference_endpoints.py

```diff
@@ -220,7 +221,7 @@ def wait(self, timeout: Optional[int] = None, refresh_every: int = 5) -> "Infere
                 )
             if self.status == InferenceEndpointStatus.RUNNING and self.url is not None:
                 # Verify the endpoint is actually reachable
-                response = get_session().get(self.url, headers=self._api._build_hf_headers(token=self._token))
+                response = get_session().get(self.health_url, headers=self._api._build_hf_headers(token=self._token))
```

Comment:
Value returned by server is a route, not a URL

```suggestion
                health_url = f"{self.url.rstrip('/')}/{self.health_route.lstrip('/')}"
                response = get_session().get(health_url, headers=self._api._build_hf_headers(token=self._token))
```

Reply:
potentially `health_url` could be an additional class property (for easier reuse)